Refactor(bg.js): Extract sub-proc logics

duangsuse opened the pull request:
- rename file zip.sh -> run-zipdist.sh
- flip if (url.protocol == "data:") condition
- flip if (!validExts.include(ext)) condition
- rename msg invalid_characters -> path_contains_invalid_characters
- inline expressions

Erisa commented on _locales/en/messages.json:
```diff
@@ -11,7 +11,7 @@
     "message": "Save"
   },
-  "invalid_characters": {
+  "path_contains_invalid_characters": {
```
Is it not a prefix, rather than a path?

duangsuse committed:
Refactor(bg.js): Extract sub-proc logics
- rename file zip.sh -> run-zipdist.sh
- flip if (url.protocol == "data:") condition
- flip if (!validExts.include(ext)) condition
- rename msg invalid_characters -> path_contains_invalid_characters
